docs-v2/validate-mdx-links.mjs

```
@@ -0,0 +1,303 @@
+  return mdxFiles;
+}
+
+// Extract links and their line numbers from MDX content
+function extractLinks(content, filePath) {
+  const links = [];
+  const lines = content.split("\n");
+  const linkRegex = /\[([^\]]+)\]\(([^)]+)\)/g;
+
+  lines.forEach((line, index) => {
+    let match;
+    while ((match = linkRegex.exec(line)) !== null) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)